# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1296373390 18000
#      Sun Jan 30 02:43:10 2011 -0500
# Node ID e3048d7fb131dbe0409b01e16b0d3e12cfc99271
# Parent  6ef3e671c87cc54c9daa74f0ce10853a1812fe80
remove ARPACK from documentation and some config files

diff --git a/ChangeLog b/ChangeLog
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,8 +1,13 @@
+2011-01-30  John W. Eaton  <jwe@octave.org>
+
+	* configure.ac: Remove code to print warning for missing ARPACK.
+	* README.MacOS: Remove arpack from list of prerequisite software.
+
 2010-01-29  Rik  <octave@nomad.inbox5.com>
 
 	* NEWS: Deprecate saveimage function.
 
 2011-01-28  Ben Abbott  <bpabbott@mac.com>
 
 	* README.MacOS: Spelling corrections.
 	* README.MacOS: Additional modifications for building Octave manually.
diff --git a/NEWS b/NEWS
--- a/NEWS
+++ b/NEWS
@@ -411,16 +411,20 @@ Summary of important user-visible change
 
       autocor  cellidx   gammai     replot
       autocov  dispatch  is_global  saveimage 
       betai    fstat     krylovb    values
 
   * For compatibility with Matlab, mu2lin (x) is now equivalent to
     mu2lin (x, 0).
 
+  * The ARPACK library is now distributed with Octave so it no longer
+    needs to be available as an external dependency when building
+    Octave.
+
 Summary of important user-visible changes for version 3.2:
 ---------------------------------------------------------
 
  ** Compatibility with Matlab graphics has been improved.
 
     The hggroup object and associated listener callback functions have
     been added allowing the inclusion of group objects.  Data sources
     have been added to these group objects such that
diff --git a/README.MacOS b/README.MacOS
--- a/README.MacOS
+++ b/README.MacOS
@@ -147,17 +147,16 @@ TODO - when a Fink package is developed 
 
 
 2.3.1 Dependencies
 ------------------
 
 To build Octave's sources the list of Fink packages below need to be installed.
 
   aquaterm
-  arpack
   autoconf (>= 2.6.0)
   automake (>= 1.11)
   fftw3 (>= 3.1.1)
   fftw3-shlibs (>= 3.1.1)
   flex >= 2.5.30
   fltk-aqua-shlibs 
   glpk-dev
   glpk-shlibs
diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -2228,21 +2228,16 @@ if test -n "$warn_cholmod"; then
   warn_msg_printed=true
 fi
 
 if test -n "$warn_cxsparse"; then
   AC_MSG_WARN([$warn_cxsparse])
   warn_msg_printed=true
 fi
 
-if test -n "$warn_arpack"; then
-  AC_MSG_WARN([$warn_arpack])
-  warn_msg_printed=true
-fi
-
 if test -n "$warn_curl"; then
   AC_MSG_WARN([$warn_curl])
   warn_msg_printed=true
 fi
 
 if test -n "$warn_fftw"; then
   AC_MSG_WARN([$warn_fftw])
   warn_msg_printed=true
diff --git a/doc/ChangeLog b/doc/ChangeLog
--- a/doc/ChangeLog
+++ b/doc/ChangeLog
@@ -1,8 +1,13 @@
+2011-01-30  John W. Eaton  <jwe@octave.org>
+
+	* interpreter/install.txi (Compiling Octave with 64-bit	Indexing):
+	Remove info about ARPACK.
+
 2011-01-29  Rik  <octave@nomad.inbox5.com>
 
 	* interpreter/octave.texi: Add new subsection on FLTK interaction.
 	* interpreter/plot.txi: Add fltk_gui_mode, fltk_mouse_wheel_zoom
 	functions to documentation.
 
 2011-01-29  Rik  <octave@nomad.inbox5.com>
 
diff --git a/doc/interpreter/install.txi b/doc/interpreter/install.txi
--- a/doc/interpreter/install.txi
+++ b/doc/interpreter/install.txi
@@ -135,19 +135,16 @@ Use <char> as the path separation charac
 running Octave on non-Unix systems.
 
 @item --without-amd
 Don't use @sc{amd}, disable some sparse matrix functionality.
 
 @item --without-camd
 Don't use @sc{camd}, disable some sparse matrix functionality.
 
-@item --without-arpack
-Don't use @sc{arpack}, disable @code{eigs} function.
-
 @item --without-colamd
 Don't use @sc{colamd}, disable some sparse matrix functionality.
 
 @item --without-ccolamd
 Don't use @sc{ccolamd}, disable some sparse matrix functionality.
 
 @item --without-cholmod
 Don't use @sc{cholmod}, disable some sparse matrix functionality.
@@ -397,66 +394,16 @@ The options @option{-fdefault-integer-8}
 CPU) have to be added to the variable @code{OPTS} and @code{NOOPT}.
 
 @item
 Once you have compiled this library make sure that you use it for
 compiling Suite Sparse and Octave.  In the following we assume that
 you installed the @sc{lapack} library as $prefix64/lib/liblapack.a.
 @end itemize
 
-@item @sc{arpack} (@url{http://www.caam.rice.edu/software/ARPACK})
-
-@itemize @minus
-@item
-In @file{ARmake.inc} set the following variables:
-
-@example
-@group
-home=path to directory @sc{arpack}
-FC=gfortran 
-FFLAGS=-fPIC -fdefault-integer-8
-MAKE=/usr/bin/make 
-ARPACKLIB=$(home)/libarpack.a
-DIRS=$(UTILdir) $(SRCdir)
-@end group
-@end example
-
-@item
-Edit the file @file{UTIL/second.f} and change the line
-
-@example
-EXTERNAL           ETIME
-@end example
-
-@noindent
-to
-
-@example
-INTRINSIC          ETIME
-@end example
-
-@item
-After building the @file{.a} library, you can make a shared version with
-
-@example
-@group
-mkdir tmp
-cd tmp
-ar x ../libarpack.a
-gcc -shared -o ../libarpack.so *.o -L$prefix64/lib \
-    -llapack -lblas 
-cd ..
-rm -rf tmp
-@end group
-@end example
-
-@noindent
-Copy the library @file{libarpack.so} to @file{$prefix64/lib/libarpack.a}.
-@end itemize
-
 @item QRUPDATE (@url{http://sourceforge.net/projects/qrupdate})
 
 In the @file{Makeconf} file:
 
 @itemize @minus
 @item
 Add @option{-fdefault-integer-8} to @env{FFLAGS}.
 
diff --git a/src/ChangeLog b/src/ChangeLog
--- a/src/ChangeLog
+++ b/src/ChangeLog
@@ -1,8 +1,18 @@
+2011-01-30  John W. Eaton  <jwe@octave.org>
+
+	* Makefile.am (OCTAVE_LIBS): Remove $(ARPACK_LDFLAGS) and
+	$(ARPACK_LIBS) from the list.
+	(DLD-FUNCTIONS/eigs.df): Don't add $(ARPACK_CPPFLAGS) to CPPFLAGS.
+	(DLD_FUNCTIONS_eigs_la_CPPFLAGS): Remove $(ARPACK_CPPFLAGS) from
+	the list.
+	(DLD_FUNCTIONS_eigs_la_LIBADD): Remove $(ARPACK_LDFLAGS) and
+	$(ARPACK_LIBS) from the list.
+
 2011-01-30  John W. Eaton  <jwe@octave.org>
 
 	* load-save.cc (Fload): Doc fix.
 
 2011-01-29  Rik  <octave@nomad.inbox5.com>
 
 	* DLD-FUNCTIONS/__init_fltk__.cc (fltk_gui_mode, fltk_mouse_wheel_zoom):
 	Improve docstrings.
diff --git a/src/Makefile.am b/src/Makefile.am
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -499,17 +499,16 @@ ALL_DEF_FILES = $(DEF_FILES) $(DLD_DYNAM
 if AMCOND_ENABLE_DYNAMIC_LINKING
   OCTAVE_LIBS = \
     ./liboctinterp.la \
     ../liboctave/liboctave.la \
     ../libcruft/libcruft.la \
     ../libcruft/libranlib.la \
     ../libgnu/libgnu.la \
     $(SPARSE_XLDFLAGS) $(SPARSE_XLIBS) \
-    $(ARPACK_LDFLAGS) $(ARPACK_LIBS) \
     $(QRUPDATE_LDFLAGS) $(QRUPDATE_LIBS) \
     $(FFTW_XLDFLAGS) $(FFTW_XLIBS) \
     $(LAPACK_LIBS) $(BLAS_LIBS) \
     $(GRAPHICS_LDFLAGS) $(GRAPHICS_LIBS) \
     $(FT2_LDFLAGS) $(FT2_LIBS) \
     $(HDF5_LDFLAGS) $(HDF5_LIBS) $(Z_LDFLAGS) $(Z_LIBS) \
     $(OPENGL_LIBS) $(X11_LIBS) $(CARBON_LIBS) \
     $(READLINE_LIBS) $(TERM_LIBS) \
@@ -526,17 +525,16 @@ else
   OCTAVE_LIBS = \
     ./liboctinterp.la \
     ../liboctave/liboctave.la \
     ../libcruft/libcruft.la \
     ../libcruft/libranlib.la \
     ../libgnu/libgnu.la \
     $(FFTW_XLDFLAGS) $(FFTW_XLIBS)
     $(QHULL_LDFLAGS) $(QHULL_LIBS) \
-    $(ARPACK_LDFLAGS) $(ARPACK_LIBS) \
     $(QRUPDATE_LDFLAGS) $(QRUPDATE_LIBS) \
     $(SPARSE_XLDFLAGS) $(SPARSE_XLIBS) \
     $(REGEX_LDFLAGS) $(REGEX_LIBS) \
     $(CURL_LDFLAGS) $(CURL_LIBS) \
     $(GLPK_LDFLAGS) $(GLPK_LIBS) \
     $(MAGICK_LDFLAGS) $(MAGICK_LIBS) \
     $(GRAPHICS_LDFLAGS) $(GRAPHICS_LIBS) \
     $(FT2_LDFLAGS) $(FT2_LIBS) \
@@ -792,19 +790,19 @@ DLD_FUNCTIONS_convhulln_la_LIBADD += $(Q
 DLD-FUNCTIONS/__delaunayn__.df: CPPFLAGS += $(QHULL_CPPFLAGS)
 DLD_FUNCTIONS___delaunayn___la_CPPFLAGS = $(AM_CPPFLAGS) $(QHULL_CPPFLAGS)
 DLD_FUNCTIONS___delaunayn___la_LIBADD += $(QHULL_LDFLAGS) $(QHULL_LIBS)
 
 DLD-FUNCTIONS/__voronoi__.df: CPPFLAGS += $(QHULL_CPPFLAGS)
 DLD_FUNCTIONS___voronoi___la_CPPFLAGS = $(AM_CPPFLAGS) $(QHULL_CPPFLAGS)
 DLD_FUNCTIONS___voronoi___la_LIBADD += $(QHULL_LDFLAGS) $(QHULL_LIBS)
 
-DLD-FUNCTIONS/eigs.df: CPPFLAGS += $(ARPACK_CPPFLAGS) $(SPARSE_XCPPFLAGS)
-DLD_FUNCTIONS_eigs_la_CPPFLAGS = $(AM_CPPFLAGS) $(ARPACK_CPPFLAGS) $(SPARSE_XCPPFLAGS)
-DLD_FUNCTIONS_eigs_la_LIBADD += $(ARPACK_LDFLAGS) $(ARPACK_LIBS) $(SPARSE_XLDFLAGS) $(SPARSE_XLIBS) $(LAPACK_LIBS) $(BLAS_LIBS)
+DLD-FUNCTIONS/eigs.df: CPPFLAGS += $(SPARSE_XCPPFLAGS)
+DLD_FUNCTIONS_eigs_la_CPPFLAGS = $(AM_CPPFLAGS) $(SPARSE_XCPPFLAGS)
+DLD_FUNCTIONS_eigs_la_LIBADD += $(SPARSE_XLDFLAGS) $(SPARSE_XLIBS) $(LAPACK_LIBS) $(BLAS_LIBS)
 
 #DLD-FUNCTIONS/qz.df DLD-FUNCTIONS/qz.lo:
 DLD_FUNCTIONS_qz_la_LIBADD += $(LAPACK_LIBS) $(BLAS_LIBS)
 
 DLD-FUNCTIONS/qr.df: CPPFLAGS += $(QRUPDATE_CPPFLAGS) $(SPARSE_XCPPFLAGS)
 DLD_FUNCTIONS_qr_la_CPPFLAGS = $(AM_CPPFLAGS) $(QRUPDATE_CPPFLAGS) $(SPARSE_XCPPFLAGS)
 DLD_FUNCTIONS_qr_la_LIBADD += $(QRUPDATE_LDFLAGS) $(QRUPDATE_LIBS) $(SPARSE_XLDFLAGS) $(SPARSE_XLIBS)
 
